### PR TITLE
Feature/quotes

### PR DESCRIPTION
**Pull request by trmdy**
This pull requests adds the remaining functionality of the quotes module.

### CODE REVIEW

**Comment by trmdy** [comment]
There are still some missing functionality, which awaits #18. It will be quickly added once #18 is merged. Until then reviews are still very much appreciated!

File: quotes/tests.py [hunk 1]
```diff
@@ -170,3 +197,191 @@ def test_vote_up_pending_fails(self):
         self.client.login(username='test', password='password')
         response = self.client.post(reverse(vote_up, kwargs={'quote_id': 1}))
         self.assertEqual(response.status_code, 404)
+
+    def test_vote_up_GET_405s(self):
```
Comment (chdefrene) [inline quotes/tests.py]: It is not immediately clear what this test wishes to accomplish, wrt. the other vote up tests. Can we rename it to make it more clear?

File: quotes/tests.py [hunk 2]
```diff
@@ -170,3 +197,191 @@ def test_vote_up_pending_fails(self):
...
+        self.client.login(username='test', password='password')
+        response = self.client.get(reverse(vote_up, kwargs={'quote_id': 2}))
+        self.assertEqual(response.status_code, 405)
```
Comment (chdefrene) [inline quotes/tests.py]: I suggest we use response status codes from the 'status' library, as we don't need to memorize status codes and meanings 😛

**Comment by chdefrene** [comment]
It may be just me, but I have trouble understanding what the tests are supposed to be doing.

I propose we follow the naming conventions at my work, where we construct test names from:

`def test_PART_TO_TEST__INITIAL_STATE_OR_INPUTS__EXPECTED_OUTCOME`

File: quotes/tests.py (hunk 1 above)
Comment (trmdy) [reply quotes/tests.py]: It tests that sending a GET request to the vote_up endpoint results in a 405 (Method not allowed). I feel it is quite clear. But I can increase the verbosity somewhat if that helps.

File: quotes/tests.py (hunk 2 above)
Comment (trmdy) [reply quotes/tests.py]: Agreed.

**Comment by trmdy** [comment]
Can you give me some specific examples? Reading over them now I feel most of them are fairly clear. But perhaps not from the method name alone. Some even have explanatory comments.

The proposed naming conventions sound reasonable (and is of course one of the more common and popular choices of test naming conventions). Personally I rarely follow any, but for a project like this I definitely see the value in it. Could you perhaps write something about it in CONTRIBUTING.md? At some point we could then go over the existing tests and rewrite them if needed. We should perhaps not enforce them absolutely, but rather as a suggestion. In my opinion, as long as the intention is fairly clear it should be sufficient.

**Comment by trmdy** [comment]
The last commit finalizes the quote app in this round; until we start implementing the actual design.

**Comment by chdefrene** [comment]
An example of the test naming I proposed:
Currently, we have the test `test_vote_up_user_has_not_voted_yet`. However, it is hard to know just from the name what this test tries to accomplish, and the words kinda melt together and makes the name hard to read. I propose renaming it to `test_vote_up__user_has_not_voted_yet__status_ok` in order to make it more readable, and to make it clear(er) what to expect from the test.

Another example is the test `test_vote_down_has_voted_down_already` that has a comment inside the function definition. In this case, we can simply rename it to `test_vote_down__user_has already_voted_down__do_not_update_sum` which makes the test much more nicer to read 😍

I'll go ahead and update the CONTRIBUTING.md file.

**Comment by trmdy** [comment]
Looks good. I will rename some of the tests.

I feel though that keeping this PR back for over a month due to tests not being properly named is nitpicking to a level we cannot afford in terms of progression.

**Comment by trmdy** [comment]
Almost all the tests have been renamed now to conform with the new guidelines, and should in general be much more readable indeed.

**Comment by chdefrene** [comment]
Good! Just for the record, it's infrequent work on this project that held this particular PR back, not intentional delays du to nit-picking

**Review by chdefrene — APPROVED** [review]
OK

**Comment by trmdy** [comment]
Yeah I chose my words badly; I did not try to imply they were intentional. Sorry about that. I was just trying to make the point that the iteration-speed could be faster if the PR was approved faster, whilst taking the discussion points we've had here outside the PR itself. (And if, as it where, improvements to the general guidelines were made, new edits could've been added retrospectively)

**Pull request merged**